controller/task_queue/handler/embedding.py

```diff
@@ -34,15 +37,30 @@ def __start_task(task: Dict[str, Any]) -> bool:
 
     user_id = task["created_by"]
     attribute_id = task["task_info"]["attribute_id"]
-    embedding_handle = task["task_info"]["embedding_handle"]
-    if task["task_info"]["embedding_type"] == EmbeddingType.ON_ATTRIBUTE.value:
-        embedding_manager.create_attribute_level_embedding(
-            project_id, user_id, attribute_id, embedding_handle
-        )
-    else:
-        embedding_manager.create_token_level_embedding(
-            project_id, user_id, attribute_id, embedding_handle
-        )
+    embedding_type = task["task_info"]["embedding_type"]
+    embedding_name = task["task_info"]["embedding_name"]
+    platform = task["task_info"]["platform"]
+    model = task["task_info"]["model"]
+    api_token = task["task_info"]["api_token"]
+
+    terms_text = task["task_info"]["terms_text"]
+    terms_accepted = task["task_info"]["terms_accepted"]
+    embedding_item = embedding_db_bo.create(
+        project_id,
+        attribute_id,
+        embedding_name,
+        enums.EmbeddingState.INITIALIZING.value,
+        type=embedding_type,
+        model=model,
+        platform=platform,
+        api_token=api_token,
+
+    )
+    agreement_db_bo.create(project_id, user_id, terms_text, terms_accepted, xfkey=embedding_item.id, xftype=enums.AgreementType.EMBEDDING.value, with_commit=True)
```

Comment:
not all embedings get an agreement entry, only cohere & openai

Reply:
or maybe only if a text and accepted != None?

Reply:
solved